source/beamline/AMBeamlineControlGroup.cpp

```diff
@@ -0,0 +1,62 @@
+#include "AMBeamlineControlGroup.h"
+#include "beamline/AMBeamlineControl.h"
+
+AMBeamlineControlGroup::AMBeamlineControlGroup(const QString &name, QObject *parent) :
+	AMConnectedControl(name, "", parent)
+{
+	status_ = new AMBeamlineControlGroupStatus(QString("%1%2").arg(name).arg("Status"), this);
+}
+
+AMBeamlineControlGroup::~AMBeamlineControlGroup()
+{
+
+}
+
+bool AMBeamlineControlGroup::isBad() const
+{
+	return status_->isBad();
+}
+
+bool AMBeamlineControlGroup::isGood() const
+{
+	return status_->isGood();
+}
+
+bool AMBeamlineControlGroup::hasControl(AMControl *control) const
+{
+	return status_->hasChildControl(control);
+}
+
+bool AMBeamlineControlGroup::addControl(AMBeamlineControl *control)
+{
+	bool result = false;
+
+	if (control && !children_.contains(control)) {
+		addChildControl(control);
```

Comment:
I agree, that's a change that definitely makes sense. We could probably also rename those methods, as we're not actually adding/removing a 'child' in the formal sense--not reparenting anything, I mean.